frontend/src/CompanyInfo.jsx

```diff
@@ -3,16 +3,126 @@ import AppSidebar from "./components/AppSidebar";
 import SearchBar from "./components/SearchBar";
 import { BASE_URL } from "./lib/utils";
 import { useState, useEffect } from "react";
-import { UserFullName } from "./context/UserContext";
+import { UserInfo } from "./context/UserContext";
 import TradingViewWidget from "./components/TradingViewWidget";
 import NewsList from "./components/NewsList";
 const logoKey = import.meta.env.VITE_LOGO_TOKEN;
+import { useParams } from "react-router-dom";
+
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  PopoverClose,
+} from "./components/ui/popover";
+
+const AddToPortfolio = ({ companyId }) => {
+  const [portfolios, setPortfolios] = useState(null);
+  const [portfoliosToAddToo, setPortfoliosToAddToo] = useState([]);
+
+  const AddToPortfolios = async () => {
+    for (let id of portfoliosToAddToo) {
+      await fetch(`${BASE_URL}/portfolios/add/${id}/${companyId}`, {
```

Comment:
From an API design perspective, it might be better to make the calls batch-able here? It's fairly small potatoes since I doubt users will add companies to THAT many portfolios at a time, but it's good practice to reduce the number of network requests wherever possible.

This would probably require implementing another API endpoint where you can provide a list of portfolios to update with companyId. Note that this raises some UX questions around what to do when one of the portfolios already has that company in it.

Reply:
+1 maybe look into `Promise.all` (https://www.geeksforgeeks.org/javascript/how-to-fetch-an-array-of-urls-with-promise-all/)